### PR TITLE
(feat) Add the Stock Rule menu entry on the stock item page, the stock rule listing and add CRUD

### PR DESCRIPTION
**Pull request by PatrickWaweru**
(feat) Add the Stock Rule menu entry on the stock item page, the stock rule listing and add CRUD

## Requirements
- [ X] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [X ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ X] My work includes tests or is validated by existing tests.

## Summary
(feat) Add the Stock Rule menu entry on the stock item page, the stock rule listing and add CRUD

## Screenshots

Screenshot from 2024-01-19 13-09-54
Screenshot from 2024-01-19 13-09-43
Screenshot from 2024-01-19 13-09-26
Screenshot from 2024-01-19 13-09-14
Screenshot from 2024-01-19 09-50-32

### CODE REVIEW

File: src/stock-items/stock-items.resource.ts [hunk 1]
```diff
@@ -308,8 +309,14 @@ export function useStockRules(filter: StockRuleFilter) {
     Error
   >(apiUrl, openmrsFetch);
 
+  // return {
+  //   items: data.data ? data.data : [],
+  //   isLoading,
+  //   isError: error,
+  // };
+
```
Comment (donaldkibet) [inline src/stock-items/stock-items.resource.ts:317]: Should probably remove commented code

File: src/stock-items/stock-items.resource.ts [hunk 2]
```diff
@@ -300,6 +300,7 @@ export function importStockItem(item: FormData) {
 // stock rules
 // getStockRules
 export function useStockRules(filter: StockRuleFilter) {
+  console.warn("Rules filter: " + JSON.stringify(filter));
```
Comment (donaldkibet) [inline src/stock-items/stock-items.resource.ts:303]:
```suggestion
  
```

File: src/stock-items/add-stock-item/stock-item-rules/stock-rules.resource.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    isNew ? "/" + item.uuid : ""
+  }`;
+  const abortController = new AbortController();
+  return openmrsFetch(apiUrl, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    signal: abortController.signal,
+    // body: { ...item, sourceType: item?.sourceType?.uuid },
```
Comment (donaldkibet) [inline src/stock-items/add-stock-item/stock-item-rules/stock-rules.resource.ts:76]:
```suggestion
   
```

**Review by jabahum — APPROVED** [review]
Thanks @PatrickWaweru

File: src/stock-items/add-stock-item/stock-item-rules/stock-rules.resource.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  const apiUrl = `ws/rest/v1/stockmanagement/stockrule/${ids[0]}${otherIds}`;
+  const abortController = new AbortController();
+  return openmrsFetch(apiUrl, {
+    method: "DELETE",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    signal: abortController.signal,
+  });
+}
```
Comment (donaldkibet) [inline src/stock-items/add-stock-item/stock-item-rules/stock-rules.resource.ts:60]: Please use readable variable names here. e.g 
```tsx 
deleteStockRule(stockRuleUuids: stringArray<string>)
 ```

hard to follow what `p c i` mean

**Review by donaldkibet — APPROVED** [review]
Thanks @PatrickWaweru left a few comments for follow up
